sqlglot/optimizer/merge_subqueries.py

```diff
@@ -181,6 +184,12 @@ def _merge_from(outer_scope, inner_scope, node_to_replace, alias):
     """
     new_subquery = inner_scope.expression.args.get("from").expressions[0]
     node_to_replace.replace(new_subquery)
+    for table_hint in outer_scope.table_hints:
+        tables = table_hint.find_all(exp.Table)
+        for table in tables:
+            if table.alias_or_name == node_to_replace.alias_or_name:
+                new_value = new_subquery.this if isinstance(new_subquery, exp.Alias) else new_subquery
```

Comment:
We only want the _alias_ of the new subquery, right? Otherwise we might end up shoving an entire derived table in here. Might want to test that.

Reply:
There isn't an example currently in the tests (and that is all the tests) where `new_subquery` comes back as anything but a table or an alias. Can you describe more the case where you think it might actually return a subquery or am I misunderstanding what you are saying?

Reply:
Try this:
```sql
SELECT 
  x.a, 
  x.c 
FROM (
  SELECT 
    x.a, 
    x.c 
  FROM (
    SELECT 
      x.a, 
      COUNT(1) AS c 
    FROM x 
    GROUP BY x.a
  ) AS x
) AS x
```

Reply:
Thanks that did it and I confirmed the test I added covers this case.